tests/kernel/kselftests.pm

```diff
@@ -14,24 +14,39 @@ use testapi;
 use warnings;
 use serial_terminal 'select_serial_terminal';
 use LTP::WhiteList;
+use version_utils qw(is_transactional);
+use transactional 'trup_install';
+
+sub download_kernel_source
+{
+    my @kv = split /\./, script_output "uname -r";
+    my ($kv0, $kv1, $kv2) = ($kv[0], $kv[1], (split /-/, $kv[2])[0]);
+    my $url = "https://mirrors.edge.kernel.org/pub/linux/kernel/v$kv0.x/linux-$kv0.$kv1";
+
+    $url .= $kv2 eq '0' ? ".tar.gz" : ".$kv2.tar.gz";
+
+    assert_script_run("mkdir /root/linux");
+    assert_script_run("curl " . $url . "| tar xz --strip-components=1 -C /root/linux", timeout => 600);
+}
 
 sub run
 {
     my ($self) = @_;
-    my $repo = get_var('LINUX_REPO', 'https://github.com/torvalds/linux');
```

Comment:
BTW I wonder why the original author used github mirror instead of git.kernel.org.

Reply:
My fault :) and github was way faster